diff --git a/cmake/pcl_find_boost.cmake b/cmake/pcl_find_boost.cmake
index fece19e15..9a90b79a7 100644
--- a/cmake/pcl_find_boost.cmake
+++ b/cmake/pcl_find_boost.cmake
@@ -49,6 +49,7 @@ if(Boost_FOUND)
   # Obtain diagnostic information about Boost's automatic linking outputted 
   # during compilation time.
   add_definitions(${Boost_LIB_DIAGNOSTIC_DEFINITIONS})
+  add_definitions(-DBOOST_ERROR_CODE_HEADER_ONLY)
   include_directories(SYSTEM ${Boost_INCLUDE_DIRS})
   link_directories(${Boost_LIBRARY_DIRS})
 endif(Boost_FOUND)
